### PR TITLE
feat(ctb): Delete `IBondManager`

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Deletes the `IBondManager` and removes the concept of a bond manager contract from the `IDisputeGame` interface. This was a leaky abstraction that didn't consider dispute games which may not want bonds, and each game may need to handle bonds differently.

Instead of a standalone bond manager adhering to the `IBondManager` interface referenced by multiple dispute games of the same type, which is less gas efficient and requires pooling money in a small number of bond managers, games should hide bonding in the creation, interaction, and resolution abstractions depending on the game's characteristics.

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #8810** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8810?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #8811** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8811?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #8816** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8816?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #8821** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8821?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #8822** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8822?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #8831** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8831?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8811?utm_source=stack-comment).

File: packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol [hunk 1]
```diff
@@ -60,9 +59,6 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
     /// @inheritdoc IDisputeGame
     GameStatus public status;
 
-    /// @inheritdoc IDisputeGame
-    IBondManager public bondManager;
```
Comment (tynes) [inline packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol:64]: This should be safe for the storage layout to change as it is a clone

File: packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol [hunk 2]
```diff
@@ -60,9 +59,6 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
...
-
     /// @inheritdoc IFaultDisputeGame
     Hash public l1Head;
```
Comment (tynes) [inline packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol:64]: nit: semver update

File: packages/contracts-bedrock/src/dispute/interfaces/IDisputeGame.sol [hunk 3]
```diff
@@ -1,7 +1,6 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.15;
 
-import { IBondManager } from "./IBondManager.sol";
 import { IInitializable } from "./IInitializable.sol";
```
Comment (tynes) [inline packages/contracts-bedrock/src/dispute/interfaces/IDisputeGame.sol]: nit: absolute path imports

**Review by tynes — APPROVED** [review]

**Pull request merged**